ethexe/signer/src/storage.rs

```diff
@@ -0,0 +1,241 @@
+use anyhow::{anyhow, bail, Result};
+use std::{collections::BTreeMap, fmt, fs, path::PathBuf, str::FromStr};
+use tempfile::TempDir;
+
+use crate::{Address, PrivateKey, PublicKey};
+
+pub trait KeyStorage: fmt::Debug + Send + Sync + 'static {
+    /// Get a private key for the public one from the key store.
+    fn get_private_key(&self, key: PublicKey) -> Result<PrivateKey>;
+
+    /// Get a public key for the provided ethereum address. If no key found a `None` is returned.
+    ///
+    /// Note: Could be very expensive cause bypasses all keys in storage.
+    fn get_key_by_addr(&self, address: Address) -> Result<Option<PublicKey>> {
+        let keys = self.list_keys()?;
+
+        for key in keys {
+            if key.to_address() == address {
+                return Ok(Some(key));
+            }
+        }
+
+        Ok(None)
+    }
+
+    /// Add a key pair to the key store.
+    fn add_key_pair(&mut self, public_key: PublicKey, private_key: PrivateKey) -> Result<()>;
```

Comment:
I think this could be removed, since its common practice to add/import key only with private key

Reply:
done